lib/debootstrap.sh

```diff
@@ -590,16 +590,37 @@ create_image()
 	cp $SDCARD/etc/armbian.txt $DESTIMG
 	mv ${SDCARD}.raw $DESTIMG/${version}.img
 
-	if [[ $COMPRESS_OUTPUTIMAGE == yes && $BUILD_ALL != yes ]]; then
-		[[ -f $DEST/images/$CRYPTROOT_SSH_UNLOCK_KEY_NAME ]] && cp $DEST/images/$CRYPTROOT_SSH_UNLOCK_KEY_NAME $DESTIMG/
-		# compress image
+	if [[ $COMPRESS_OUTPUTIMAGE == yes ]]; then
+		COMPRESS_OUTPUTIMAGE="sha,7z"
+	fi
+
+	if [[ $COMPRESS_OUTPUTIMAGE == *sha* && $BUILD_ALL != yes ]]; then
 		cd $DESTIMG
+		display_alert "SHS256 calculating" "${version}.img" "info"
 		sha256sum -b ${version}.img > sha256sum.sha
+		cp sha256sum.sha "$DEST/images/${version}.img.sha"
 		if [[ -n $GPG_PASS ]]; then
-			echo $GPG_PASS | gpg --passphrase-fd 0 --armor --detach-sign --pinentry-mode loopback --batch --yes ${version}.img
+			echo $GPG_PASS | gpg --passphrase-fd 0 --armor --detach-sign --pinentry-mode loopback --batch --yes ${version}.img || exit 1
 		fi
-			display_alert "Compressing" "$DEST/images/${version}.img" "info"
-		7za a -t7z -bd -m0=lzma2 -mx=3 -mfb=64 -md=32m -ms=on $DEST/images/${version}.7z ${version}.key ${version}.img armbian.txt *.asc sha256sum.sha >/dev/null 2>&1
+		cd ..
+	fi
+	#
+	if [[ $COMPRESS_OUTPUTIMAGE == *7z* && $BUILD_ALL != yes ]]; then
+		[[ -f $DEST/images/$CRYPTROOT_SSH_UNLOCK_KEY_NAME ]] && cp $DEST/images/$CRYPTROOT_SSH_UNLOCK_KEY_NAME $DESTIMG/
+		# compress image
+		cd $DESTIMG
+		display_alert "Compressing" "$DEST/images/${version}.7z" "info"
+		gz 7za a -t7z -bd -m0=lzma2 -mx=3 -mfb=64 -md=32m -ms=on $DEST/images/${version}.7z ${version}.key ${version}.img armbian.txt *.asc sha256sum.sha >/dev/null 2>&1
+		cd ..
+	fi
+	#
+	if [[ $COMPRESS_OUTPUTIMAGE == *gz* && $BUILD_ALL != yes ]]; then
+		display_alert "Compressing" "$DEST/images/${version}.img.gz" "info"
+		gzip < $DESTIMG/${version}.img > $DEST/images/${version}.img.gz
+	fi
+	if [[ $COMPRESS_OUTPUTIMAGE == *bz2* && $BUILD_ALL != yes ]]; then
```

Comment:
Do we need bz2?

Please update https://github.com/armbian/documentation/blob/master/docs/Developer-Guide_Build-Options.md after we merge this.

Reply:
Another small improvement would be to use pigz for compression. Add it here: https://github.com/armbian/build/blob/master/lib/general.sh#L549

Reply:
pigz rules! We can put bz2 into trash with pigz performance.

Reply:
OK, make those changes and I think we are ready to go. According to our new policy, we have to wait for second approval.